packages/salesforcedx-vscode-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -67,27 +86,55 @@ export class ApexReplayDebug extends DebugSession {
     response: DebugProtocol.LaunchResponse,
     args: LaunchRequestArguments
   ): void {
+    this.setupLogger(args);
+
+    this.log(
+      TRACE_CATEGORY_LAUNCH,
+      `launchRequest: args=${JSON.stringify(args)}`
+    );
     if (args.logFile) {
       args.logFile = this.convertDebuggerPathToClient(args.logFile);
     }
-    this.logContext = new LogContext(args);
+    this.logContext = new LogContext(args, this.breakpointUtil);
     if (!this.logContext.hasLogLines()) {
       response.success = false;
       response.message = nls.localize('no_log_file_text');
       this.sendResponse(response);
       return;
     }
-    if (args.stopOnEntry) {
-      this.logContext.updateFrames();
-      this.sendEvent(new StoppedEvent('entry', ApexReplayDebug.THREAD_ID));
-    } else {
-      // Continue until first breakpoint
-    }
     this.printToDebugConsole(
       nls.localize('session_started_text', this.logContext.getLogFileName())
     );
     response.success = true;
     this.sendResponse(response);
+
+    if (args.stopOnEntry) {
+      // Stop in the debug log
+      this.logContext.updateFrames(this.getHandlerForDebugConsole());
+      this.sendEvent(new StoppedEvent('entry', ApexReplayDebug.THREAD_ID));
+    } else {
+      // Set breakpoints first, then try to continue to the next breakpoint
+      setTimeout(() => {
```

Comment:
After `launchRequest`, we want VS Code to send breakpoints to the adapter first before we force a continue. This timeout is to delay the forced continue.

Reply:
I reread the debug api doc and this approach seems kind of weird. Maybe I'm misreading the docs but it looks like this would go into configurationDoneRequest? Wouldn't that send all the breakpoint info to the adapter first and then we can send the continueRequest once the breakpoint configuration is done?
https://code.visualstudio.com/docs/extensionAPI/api-debugging

Reply:
Thanks, I forgot about `configurationDoneRequest`. Will update.